src/firebolt/utils/exception.py

```diff
@@ -267,3 +270,37 @@ class NotSupportedError(DatabaseError):
 
 class ConfigurationError(InterfaceError):
     """Invalid configuration error."""
+
+
+class FireboltStructuredError(Error):
+    """Base class for structured errors received in JSON body."""
+
+    # Output will look like this after formatting:
+    # "{severity}: {name} ({code}) - {description}, see {helpLink}"
+    message_template = "{severity}{name}{code}{description}{at}{helpLink}"
+
+    def __init__(self, json: Dict[str, Any]):
+        self.json = json
+        self.errors = json.get("errors", [])
+
+    def __str__(self) -> str:
+        error_messages = []
+        for error in self.errors:
+            severity = f"{error['severity']}: " if error.get("severity") else ""
+            name = f"{error['name']} " if error.get("name") else ""
+            code = f"({error['code']}) " if error.get("code") else ""
+            description = (
+                f"- {error['description']}" if error.get("description") else ""
+            )
+            helpLink = f", see {error['helpLink']}" if error.get("helpLink") else ""
+            at = f" at {error['location']}" if error.get("location") else ""
+            message = self.message_template.format(
+                severity=severity,
```

Comment:
This is a personal code style preference that shouldn't be a part of a code review.
Specifically to avoid this we have a linter. This syntax is compliant.

Reply:
Sure but in other places you write spaces. I just thought that you forgot them here. I also think spaces before and after `=` improve the code readability and make other people to easier search in the code. If for example I want to look for the pace(s) where variable `foo` is assigned I am looking for `foo =`, not `foo=`. Yes, I know about search using regex, but IMHO typically people use regular search. BTW this discussion reminded me mine discussion that happened in 1999 (previous century!). That time I advocated your position about personal coding style. Since that I have changed my mind :).

Reply:
https://peps.python.org/pep-0008/#whitespace-in-expressions-and-statements